token/client/src/token.rs

```diff
@@ -636,6 +637,14 @@ where
             }
         }
 
+        // ensure the blockhash is still the latest one
+        if self.nonce_blockhash.is_none() {
+            blockhash = self
+                .client
+                .get_latest_blockhash()
+                .await
+                .map_err(TokenError::Client)?
+        }
```

Comment:
To go along with my other suggestion, we can change this to:

```suggestion
        let blockhash = if let Some(blockhash) = self.nonce_blockhash {
            blockhash
        } else {
            self.client
                .get_latest_blockhash()
                .await
                .map_err(TokenError::Client)?
        };
```